### DISCUSSION TITLE
How to stream the tool invocation steps to the client for display

### DISCUSSION THREAD

**Discussion by kalcifield** [spring-projects/spring-ai#1908] [Q&A]
I am curious if its possible to stream the function calls in the Assistant response right before doing it, so the user knows what process is going on in the background. Like in the case of perplexity:
![image](https://github.com/user-attachments/assets/1a76003d-7bf2-4a5c-af7a-0284700070d0)
I am using the bedrock converse implementation.
I suppose the proxy toolcall might be of use, but I would prefer to extend the original assistant message on the fly

**Comment by henningSaulCM** [comment]
Hi @kalcifield, just fyi, I had a similar question a while back: https://github.com/spring-projects/spring-ai/discussions/960 (but no response/update).
